powerdns_sync/lib/Atomia/DNS/PowerDNSSyncer.pm

```diff
@@ -119,6 +119,11 @@ sub reload_updated_zones {
 	my $zonestatusarray = $self->soap->GetZoneStatusBulk($changes_to_keep_name);
 	$zonestatusarray = $zonestatusarray->result;
 
+	my $zone_status_hash = {};
+	foreach my $zone_status (@$zonestatusarray) {
+		$zone_status_hash->{$zone_status->{'zonename'}} = $zone_status->{'zonestatus'};
```

Comment:
Here we should make sure that $zone_status->{'zonename'} cannot be undefined.

Reply:
`$zone_status->{'zonename'}` shouldn't be undefended anyway. This part of code seems to go trough the array retuned by `GetZoneStatusBulk` and create a hash `zone_status_hash` using its values. The only way `$zone_status->{'zonename'}` can be undefined is if the array is empty, in which case the foreach loop will skip creating the hash.